cryptlib.c

```diff
@@ -53,6 +53,14 @@ static void cryptodev_complete(struct crypto_async_request *req, int err)
 	complete(&res->completion);
 }
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0))
```

Comment:
API is updated in 6.2.0 kernel. I think this should be 6 , 2, 0

Reply:
The API change first appeared in v6.3-rc1 and has not been backported to v6.2. Compare the definition of `crypto_completion_t` in:

* Linux 6.2.6: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/linux/crypto.h?h=v6.2.6#n179
* Linux 6.3-rc2: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/crypto.h?h=v6.3-rc2#n179

I also build cryptodev with this patch against both 6.2.x and 6.3-rcX on my system and it works fine in both cases.

Reply:
I generally avoid macros having the same name as functions but if there are no other review questions we'll merge this patch in the next days

Reply:
you are right. thanks for correcting me.

Reply:
> I generally avoid macros having the same name as functions but if there are no other review questions we'll merge this patch in the next days

It's a bit dirty either way, either the core code/logic gets polluted with `#ifdefs`, or the compatibility logic is well-separated but relies on macro trickery to "hijack itself in". In this case I opted for the second approach but I'm never sure which is the best (as in: less worst) approach of both.

Either way, thanks for merging!